lib/obj2gltf.js

```diff
@@ -78,64 +79,71 @@ function obj2gltf(objPath, gltfPath, options) {
+        extension = '.glb';
     }
 
     gltfPath = path.join(path.dirname(gltfPath), modelName + extension);
-    var resourcesDirectory = options.bypassPipeline ? path.dirname(gltfPath) : obj2gltf._getTempDirectory();
-
-    var aoOptions = ao ? {} : undefined;
-    var kmcOptions = kmc ? {} : undefined;
-
-    var pipelineOptions = {
-        createDirectory : false,
-        basePath : resourcesDirectory,
-        binary : binary,
-        embed : !separate,
-        embedImage : !separateTextures,
-        quantize : compress,
-        compressTextureCoordinates : compress,
-        encodeNormals : compress,
-        preserve : !optimize,
-        optimizeForCesium : optimizeForCesium,
-        smoothNormals : generateNormals,
-        aoOptions : aoOptions,
-        kmcOptions : kmcOptions,
-        textureCompressionOptions : textureCompressionOptions
+
+    var jsonOptions = {
+        spaces : 2
     };
 
-    return loadObj(objPath, options)
+    return loadOverridingImages(options)
+        .then(function() {
+            return loadObj(objPath, options);
+        })
         .then(function(objData) {
             return createGltf(objData, options);
         })
         .then(function(gltf) {
-            return writeUris(gltf, gltfPath, resourcesDirectory, options);
+            return writeUris(gltf, gltfPath, options);
         })
         .then(function(gltf) {
-            if (bypassPipeline) {
-                return fsExtra.outputJson(gltfPath, gltf);
+            if (binary) {
+                var glb = gltfToGlb(gltf);
+                return fsExtra.outputFile(gltfPath, glb);
             }
-            return GltfPipeline.processJSONToDisk(gltf, gltfPath, pipelineOptions);
-        })
-        .finally(function() {
-            return cleanup(resourcesDirectory, options);
+            return fsExtra.outputJson(gltfPath, gltf, jsonOptions);
         });
 }
 
-function cleanup(resourcesDirectory, options) {
-    if (!options.bypassPipeline && options.separate) {
-        fsExtra.remove(resourcesDirectory, function () {
-            // Don't fail simply because we couldn't
-            // clean up the temporary files.
-        });
-    }
+function loadOverridingImages(options) {
+    // The texture paths supplied in the .mtl may be overriden by the texture path supplied in options
+    var checkTransparencyOptions = {
+        checkTransparency : options.checkTransparency
+    };
+    var imagePaths = [options.metallicRoughnessOcclusionTexture, options.specularGlossinessTexture, options.occlusionTexture, options.normalTexture, options.baseColorTexture, options.emissiveTexture];
+    imagePaths = imagePaths.filter(function(imagePath) {return defined(imagePath);});
+    return Promise.map(imagePaths, function(imagePath) {
+        var imageOptions = (imagePath === options.baseColorTexture) ? checkTransparencyOptions : undefined;
+        return loadImage(imagePath, imageOptions);
+    }).then(function(images) {
```

Comment:
Include a concurrency limit in case someone throws an obj with a very large number of textures at this.